docs/SPEC.md

```diff
@@ -1086,6 +1078,37 @@ A package may reassign a new value to an option identifier declared in one of it
 A package cannot access nor modify the identifiers belonging to the imported packages of its imported packages.
 Every statement contained in an imported package is evaluated.
 
+### Attributes
+
+Attributes define a set of properties on the subsequent source code element.
+
+    Attributes             = { Attribute } .
+    Attribute              = "@" identifier AttributeParameters .
+    AttributeParameters    = "(" [ AttributeParameterList [ "," ] ] ")" .
```

Comment:
Thinking about this more, I think this should be forced to be a property list. At the moment, we wouldn't be able to update the edition attribute without it being its own breaking change because it takes a string literal. Maybe we should do it like:

```
@edition(name: "2022.1")
```

Thoughts?

Reply:
Hmm still thinking, my initial thoughts are that it make the edition specifically quite verbose. Also once we introduce the first edition parsing we can't change it since we need to support old parsers being able to understand it.

Reply:
Maybe it's fine as-is? Since object literals exist we can still support the property lists and the parser just needs to be very flexible.

I'm going to say it's fine as-is now that I think about it. I was also going to propose removing the commas to have multiple attribute parameters, but let's keep that in the grammar for extra flexibility and then never use it.